File: tests/automation/pages/settings_page.py
from playwright.sync_api import Page


class SettingsPage:
    def __init__(self, page: Page, base_url: str):
        self.page = page
        self.base_url = base_url

    def goto(self):
        self.page.goto(f"{self.base_url}/settings")
        self.page.wait_for_load_state("networkidle")

    @property
    def page_heading(self):
        return self.page.get_by_role("heading", name="Settings").first

    @property
    def profile_section(self):
        return self.page.locator("text=Profile").first

    @property
    def notification_section(self):
        return self.page.locator("text=Notifications").first

    @property
    def api_keys_section(self):
        return self.page.locator("text=API Keys").first

    @property
    def team_section(self):
        return self.page.locator("text=Team").first

    @property
    def save_button(self):
        return self.page.get_by_role("button", name="Save").first

File: tests/automation/pages/SettingsPage.ts
import type { Page, Locator } from '@playwright/test';
import { expect } from '@playwright/test';
import { BasePage } from './BasePage.js';

export class SettingsPage extends BasePage {
  constructor(page: Page, baseURL: string) {
    super(page, baseURL);
  }

  get apiKeySection(): Locator {
    return this.page.locator('[data-testid="api-keys-section"]');
  }

  get generateApiKeyButton(): Locator {
    return this.page.getByRole('button', { name: /generate|create.*key/i });
  }

  get notificationToggles(): Locator {
    return this.page.locator('[role="switch"], input[type="checkbox"]');
  }

  get saveButton(): Locator {
    return this.page.getByRole('button', { name: /save|update/i });
  }

  get successMessage(): Locator {
    return this.page.locator('[data-testid="success-message"], [role="alert"]').first();
  }

  async navigateTo(): Promise<void> {
    await this.goto('/settings');
    await this.waitForLoad();
  }

  async toggleNotification(index: number): Promise<void> {
    const toggle = this.notificationToggles.nth(index);
    await this.click(toggle, `Notification toggle at index ${index}`);
  }

  async clickSave(): Promise<void> {
    await this.click(this.saveButton, 'Save settings button');
  }

  async expectSettingsLoaded(): Promise<void> {
    await expect(this.page).toHaveTitle(/settings/i);
  }

  async expectSaveSuccess(): Promise<void> {
    await expect(this.successMessage).toBeVisible({ timeout: 10000 });
  }
}
